plugins/utils/config/netconfigutils_test.go

```diff
@@ -0,0 +1,105 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// +build !integration_test,!e2e_test
+
+package config
+
+import (
+	"fmt"
+	"github.com/coreos/go-iptables/iptables"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsValidPortWithValidPort(t *testing.T) {
+	ports := []string{"1", "65535", "1337", "311", " 1"}
+	for _, port := range ports {
+		result := IsValidPort(port)
+		assert.NoError(t, result)
+	}
+}
+
+func TestIsValidPortWithInvalidPort(t *testing.T) {
+	ports := []string{"a", "1*ab", "-1", "1.1", "0", "65536", "", " "}
+	for _, port := range ports {
+		result := IsValidPort(port)
+		assert.Equal(t, fmt.Sprintf("invalid port [%s] specified", port), result.Error())
+	}
+}
+
+func TestIsValidPortRangeWithValidPort(t *testing.T) {
+	ports := []int{1, 8080, 65535}
+	for _, port := range ports {
+		result := IsValidPortRange(port)
+		assert.NoError(t, result)
+	}
+}
+
+func TestIsValidPortRangeWithInValidPort(t *testing.T) {
+	ports := []int{-200, -1, 0, 65536, 90000}
+	for _, port := range ports {
+		result := IsValidPortRange(port)
+		assert.Equal(t, fmt.Sprintf("invalid port [%d] specified", port), result.Error())
+	}
+}
+
+func TestIsValidIPAddressOrCIDR(t *testing.T) {
+	type IPAddr struct {
+		ip            string
+		expectedProto iptables.Protocol
+		expectedValid bool
+	}
+	ips := []IPAddr{
+		{ip: "216.3.128.12", expectedProto: iptables.ProtocolIPv4, expectedValid: true},
+		{ip: "2001:0db8:85a3:0000:0000:8a2e:0370:7334", expectedProto: iptables.ProtocolIPv6, expectedValid: true},
+		{ip: "2001:0db8::0000:0000:8a2e:0370:7334", expectedProto: iptables.ProtocolIPv6, expectedValid: true},
+		{ip: "216.3.128.12/24", expectedProto: iptables.ProtocolIPv4, expectedValid: true},
+		{ip: "2001:0db8:85a3:0000:0000:8a2e:0370:7334/32", expectedProto: iptables.ProtocolIPv6, expectedValid: true},
+		{ip: "a", expectedProto: iptables.ProtocolIPv4, expectedValid: false},
+		{ip: "1*ab", expectedProto: iptables.ProtocolIPv4, expectedValid: false},
+		{ip: "123.222.333.444/24", expectedProto: iptables.ProtocolIPv4, expectedValid: false},
+		{ip: "123.222..444", expectedProto: iptables.ProtocolIPv4, expectedValid: false},
+		{ip: "123.222/333.444", expectedProto: iptables.ProtocolIPv4, expectedValid: false},
+	}
+	for _, ip := range ips {
+		proto, valid := IsValidIPAddressOrCIDR(ip.ip)
+		assert.Equal(t, ip.expectedProto, proto)
+		assert.Equal(t, ip.expectedValid, valid)
+	}
+}
+
+func TestIsValidCIDR(t *testing.T) {
```

Comment:
could we also test invalid CIDRS?

Reply:
This test includes both valid and invalid values.